parsing/extensions.ml

```diff
@@ -240,144 +258,70 @@ module Immutable_arrays = struct
   type nonrec pattern =
     | Iapat_immutable_array of pattern list
 
+  let extension_string = Clflags.Extension.to_string Immutable_arrays
+
   let expr_of ~loc = function
-    | Iaexp_immutable_array elts -> Ast_helper.Exp.array ~loc elts
+    | Iaexp_immutable_array elts ->
+      Expression.make_extension ~loc [extension_string] @@
```

Comment:
What you suggest is doable. But the `extension_string` is different for each extension, as is the thing after the `@@` and its type. Trying to abstract this means building up some mechanism where we specify the `extension_string` and the type and contents of the main payload. That will surely take more than one line! So because this step is so cheap (unlike the dual case in `ast_of`, where there's lots of pattern-matching and error checking), we just repeat it.

Reply:
I spent some time thinking about this and I couldn't come up with a good solution.  I concede to you here!  Could we move the explanatory comment somewhere other than "in the middle of comprehensions"?  I think it should be more discoverable -- maybe a [Note] that we reference from the various call sites?